### PR TITLE
algorithms/calorimetry/CalorimeterIslandCluster: pass PODIO collections instead of std::vector

### PR DESCRIPTION
**Pull request by veprbl**
### Briefly, what does this PR introduce?

Follow up to #741 

### What kind of change does this PR introduce?
- [x] Bug fix (issue #__)
- [ ] New feature (issue #__)
- [ ] Documentation update
- [ ] Other: __

### Please check if this PR fulfills the following:
- [ ] Tests for the changes have been added
- [ ] Documentation has been added / updated
- [ ] Changes have been communicated to collaborators

### Does this PR introduce breaking changes? What changes might users need to make to their code?
No

### Does this PR change default behavior?
No

### CODE REVIEW

File: src/algorithms/calorimetry/CalorimeterIslandCluster.cc [hunk 1]
```diff
@@ -20,19 +20,66 @@
...
+//TODO:Reconcile edm4hep::Vector2f and edm4eic::Vector3f especially with regards to the operators and sign convention
+static edm4hep::Vector2f localDistXY(const CaloHit &h1, const CaloHit &h2) {
+  //edm4eic::Vector3f h1_pos=geo_converter->position(h1.getCellID());
+  //edm4eic::Vector3f h2_pos=geo_converter->position(h2.getCellID());
+  const auto delta =h1.getLocal() - h2.getLocal();
+  return {delta.x, delta.y};
+  //const auto deltax = h1.getLocal()[0] - h2.getLocal()[0];
+  //const auto deltay = h1.getLocal()[1] - h2.getLocal()[1];
+  //return {delta.x, delta.y,0};
+  //return {deltax,deltay};
```
Comment (wdconinc) [inline src/algorithms/calorimetry/CalorimeterIslandCluster.cc:36]: Did you want to keep this around for the `TODO` above?

File: src/algorithms/calorimetry/CalorimeterIslandCluster.cc [hunk 2]
```diff
@@ -184,53 +231,44 @@ void CalorimeterIslandCluster::AlgorithmInit(std::shared_ptr<spdlog::logger>& lo
...
+std::unique_ptr<edm4eic::ProtoClusterCollection> CalorimeterIslandCluster::AlgorithmProcess(const edm4eic::CalorimeterHitCollection &hits) {
     // group neighboring hits
-    std::vector<std::vector<std::pair<uint32_t, const CaloHit*>>> groups;
-
-    //FIXME: protocluster collection to this?
-    std::vector<edm4eic::ProtoCluster> proto;
+    std::vector<std::set<std::size_t>> groups;
 
     std::vector<bool> visits(hits.size(), false);
     //TODO: use the right logger
```
Comment (wdconinc) [inline src/algorithms/calorimetry/CalorimeterIslandCluster.cc:244]:
```suggestion
```

**Review by wdconinc — DISMISSED** [review]

**Review by wdconinc — DISMISSED** [review]

File: src/algorithms/calorimetry/CalorimeterIslandCluster.cc (hunk 1 above)
Comment (veprbl) [reply src/algorithms/calorimetry/CalorimeterIslandCluster.cc:36]: I'm not sure what those comments are about. We probably don't need to consider those TODOs.

Comment (wdconinc) [reply src/algorithms/calorimetry/CalorimeterIslandCluster.cc:36]: I'd remove these comments while we're doing cleanup.

Comment (veprbl) [reply src/algorithms/calorimetry/CalorimeterIslandCluster.cc]: Done

**Review by wdconinc — APPROVED** [review]

**Pull request merged**